fe/fe-core/src/main/java/com/starrocks/catalog/HiveView.java

```diff
@@ -16,108 +16,55 @@
 
 import java.util.List;
 
 import static java.util.Objects.requireNonNull;
 
-public class HiveView extends Table {
-    private static final Logger LOG = LogManager.getLogger(HiveView.class);
-
+public class HiveView extends ConnectorView {
     public enum Type {
         Trino,
         Hive
     }
 
-    private final String catalogName;
-    private final String dbName;
-    private final String inlineViewDef;
-    private final Type viewType;
-
     public static final String PRESTO_VIEW_PREFIX = "/* Presto View: ";
     public static final String PRESTO_VIEW_SUFFIX = " */";
+    private final HiveView.Type viewType;
 
-    public HiveView(long id, String catalogName, String dbName, String name, List<Column> schema, String definition,
-                    Type type) {
-        super(id, name, TableType.HIVE_VIEW, schema);
-        this.catalogName = requireNonNull(catalogName, "Hive view catalog name is null");
-        this.dbName = requireNonNull(dbName, "Hive view db name is null");
-        this.inlineViewDef = requireNonNull(definition, "Hive view text is null");
+    public HiveView(long id, String catalogName, String dbName, String name, List<Column> schema, String definition, Type type) {
+        super(id, catalogName, dbName, name, schema, definition, TableType.HIVE_VIEW);
         this.viewType = requireNonNull(type, "Hive view type is null");
     }
 
-    public QueryStatement getQueryStatementWithTrinoParser(SessionVariable sessionVariable)
-            throws StarRocksPlannerException {
-        sessionVariable.setSqlDialect("trino");
-        return getQueryStatementImpl(sessionVariable);
-    }
-
-    public QueryStatement getQueryStatementWithDefaultParser(SessionVariable sessionVariable)
-            throws StarRocksPlannerException {
-        return getQueryStatementImpl(sessionVariable);
-    }
-
-    private QueryStatement getQueryStatementImpl(SessionVariable sessionVariable) {
-        ParseNode node;
-        try {
-            node = com.starrocks.sql.parser.SqlParser.parse(inlineViewDef, sessionVariable).get(0);
-        } catch (Exception e) {
-            LOG.warn("stmt is {}", inlineViewDef);
-            LOG.warn("exception because: ", e);
-            if (viewType == Type.Trino) {
-                // try to parse with starrocks sql dialect
-                sessionVariable.setSqlDialect("starrocks");
-                node = com.starrocks.sql.parser.SqlParser.parse(inlineViewDef, sessionVariable).get(0);
-            } else {
-                throw new StarRocksPlannerException(
-                        String.format("Failed to parse view-definition statement of view: %s", name),
-                        ErrorType.INTERNAL_ERROR);
-            }
+    @Override
+    public QueryStatement doGetQueryStatement(SessionVariable sessionVariable) throws StarRocksPlannerException {
+        if (viewType == HiveView.Type.Trino) {
+            sessionVariable.setSqlDialect("trino");
```

Comment:
can we add class TrinoView extends ConnectorView? so we do not need to do like this